compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/PersistRecordValidator.java

```diff
@@ -110,15 +116,27 @@ public void perform(SyntaxNodeAnalysisContext ctx) {
                     return;
                 }
             }
+            Node node = ctx.node();
+            if (node instanceof EnumDeclarationNode) {
+                EnumDeclarationNode enumDeclarationNode = (EnumDeclarationNode) node;
+                Optional<MetadataNode> metadata = enumDeclarationNode.metadata();
+                metadata.ifPresent(metadataNode -> checkPresenceOfEntityAnnotation(ctx, metadataNode.annotations(),
+                        DiagnosticsCodes.PERSIST_128));
+                return;
+            } else if (node instanceof TypeCastExpressionNode) {
+                NodeList<AnnotationNode> annotations = ((TypeCastExpressionNode) node).typeCastParam().annotations();
+                checkPresenceOfEntityAnnotation(ctx, annotations, DiagnosticsCodes.PERSIST_129);
```

Comment:
`checkPresenceOfEntityAnnotation` is doing two function,
1. Check for entity annotation
2. If present report diagnostics

IMO it should only do the first function. `isEnitityAnnotationPresent`. Based on the flag return report diagnostics should be called from main function. This makes it easier to grab the logic at this level also reduced passing unnecessary params to function